### PR TITLE
Doc `INCLUDE` clause for sources

### LINKED ISSUE #1805: Document: feat: only ingest key-ed value in additional header column
This issue tracks the documentation update needed for the merged PR #14628.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/14628
Source PR Merged At: 2024-01-25T05:04:48Z

### PR DESCRIPTION
**Pull request by emile-00**
## Info

- **Description**

  - Doc `INCLUDE` clause for supported sources. Create separate page for thorough explanations. 
  - Let me know if it's better have the info consolidated onto one page or if each connector guide should have its own section instead.

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - https://github.com/risingwavelabs/risingwave/pull/14215, https://github.com/risingwavelabs/risingwave/pull/14628

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/1805, https://github.com/risingwavelabs/risingwave-docs/issues/1750

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/ingest/include-clause.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+|--------------------|--------------------------------------------|---------------------------------------------------|
+| key                | `bytea`                                    | Can be overwritten by `ENCODE` and `KEY ENCODE`.  |
+| timestamp          | `timestamp with time zone` (i64 in millis) | Refer to `CreateTime` rather than `LogAppendTime`.|
+| partition          | `varchar`                                  | The partition the message is from.                |
+| offset             | `varchar`                                  | The offset in the partition.                      |
+| headers            | `struct<varchar, bytea>[]`                 | Key-value pairs along with the message.           |
+
+In the case of headers, there are two ways to define it.
+
+You can choose to generate headers with type `List[Struct<Varchar, Bytea>]`, which can be difficult to interpret in SQL.
```
Comment (tabVersion) [inline docs/ingest/include-clause.md]: maybe we not mention `which can be difficult to interpret in SQL` here. It makes users confusing why the dev team makes such a design.

File: docs/ingest/include-clause.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+```sql
+INCLUDE header [AS kafka_header]
+```
+
+Or you can generate a type `bytea` header, where the column content will be specified as the value associated with the specified key, `header_col`. In this case, the generated column name will have the format `_rw_kafka_header_{header col name}_{col type}`, where `col type` is the data type of the header column.
+
+```sql
+INCLUDE header 'header_col' [AS kafka_header]
+```
```
Comment (tabVersion) [inline docs/ingest/include-clause.md:65]: please mention that the grammar only works with `header`

**Review by tabVersion — APPROVED** [review]
rest LGTM

**Comment by tabVersion** [comment]
Is the doc released together with v1.7.0?

**Comment by hengm3467** [comment]
> Is the doc released together with v1.7.0?

@tabVersion Yes. That's the plan.

**Review by hengm3467 — APPROVED** [review]
LGTM

**Pull request merged**